prowler/providers/azure/azure_provider.py

```diff
@@ -244,23 +258,25 @@ def validate_arguments(
             SystemExit: If browser authentication is enabled but the tenant ID is not provided.
             SystemExit: If tenant ID is provided but browser authentication is not enabled.
         """
-        if (
+        if not browser_auth and tenant_id:
+            raise AzureTenantIDNoBrowserAuthError(
+                file=os.path.basename(__file__),
+                original_exception="Azure Tenant ID (--tenant-id) is required only for browser authentication mode",
```

Comment:
Why this? We should keep that exception, it is a validation one.

Reply:
I see the same code at line 276, please check it.

Reply:
I just moved it above, otherwise this error will never raise. If we check all the attributes first it will raise the `AzureNoAuthenticationMethodError` because by default, all auth methods will have the False values.

Reply:
It won't change the behaviour, I think that is better now